typed/clj.analyzer/src/typed/clj/analyzer/utils.cljc

```diff
@@ -173,23 +175,35 @@
 (defmethod maybe-class String [s]
   (maybe-class (symbol s)))
 
+(defn maybe-array-class-sym [x]
+  (let [sname (name x)]
+    (if-let [c (and (= (count sname) 1)
+                    (Character/isDigit (first sname))
+                    (namespace x))]
+      (when-let [c (or (specials c)
+                       (maybe-class-from-string c))]
+        (array-class (Integer/parseInt sname) c)))))
+
 (defmethod maybe-class Symbol [sym]
-  (when-not (namespace sym)
-    (let [sname (name sym)
-          snamec (count sname)]
-      (if-let [base-type (and (#?(:cljr .EndsWith :default .endsWith) sname "<>")
-                              (maybe-class (subs sname 0 (- snamec 2))))]
-        (array-class base-type)
-        (if-let [ret (or (specials sname)
-                         (special-arrays sname))]
-          ret
-          (maybe-class-from-string sname))))))
+  (let [sname (name sym)
+        snamec (count sname)]
+    (or (maybe-array-class-sym sym)
+        (when-not (namespace sym)
+          (if-let [base-type (and (#?(:cljr .EndsWith :default .endsWith) sname "<>")
+                                  (maybe-class (subs sname 0 (- snamec 2))))]
+            ;; TODO: we're leaking into the syntax
+            (array-class base-type)
```

Comment:
Hmm I have no idea what why this case is here, I don't think `<>` is valid syntax?

Reply:
Guessing it's something to do with `clojure.reflect`.